tests/RedisTest.php

```diff
@@ -53,19 +51,51 @@ protected function mstime() {
         return round(microtime(true)*1000);
     }
 
+    protected function getAuthParts(&$user, &$pass) {
+        $user = $pass = NULL;
+
+        $auth = $this->getAuth();
+        if ( ! $auth)
+            return;
+
+        if (is_array($auth)) {
+            if (count($auth) > 1) {
+                list($user, $pass) = $auth;
+            } else {
+                $pass = $auth[0];
+            }
+        } else {
+            $pass = $auth;
+        }
+    }
+
+    protected function getAuthFragment() {
+        $this->getAuthParts($user, $pass);
+
+        if ($user && $pass) {
+            return "auth[]=$user&auth[]=$pass";
```

Comment:
It would be great if users can set `auth[pass]=$pass&auth[user]=$user` and don't think about ordering